### PR TITLE
Add post-processing for cleaner output corpora, support for entity min_count in gensim model creation, redirects resolution

### PR DESCRIPTION
**Pull request by phdowling**
These are some changes I made that seem to improve corpus and model quality. I also added a script that converts the vectors to a csv format that can be used by Spotlight to create a vector store.

### CODE REVIEW

File: resources/fix_corpus.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    line_parts.append(edited)
+                    line_parts.append(match.group(0))
+                    start_at = match.end()
+
+                edited = line[start_at: -1].translate(replace_punctuation)
+                line_parts.append(edited)
+
+                line = "".join(line_parts)
+
+                #line = utils.to_unicode(line)
```
Comment (dav009) [inline resources/fix_corpus.py]: comment

File: src/main/scala/org/idio/wikipedia/redirects/RedirectStore.scala [hunk 2]
```diff
@@ -55,14 +55,27 @@ class RedisRedirectStore(redisHost:String, redisPort:Int, redirects:Iterator[(St
...
             val matches = entityRegex.findAllIn(line).matchData
-            val origin = URLDecoder.decode(matches.next().group(1), "utf-8")
-            val destination = URLDecoder.decode(matches.next().group(1), "utf-8")
-            //print(origin + "-->" + destination)
-            (origin, destination)
+            if(matches.hasNext){
+                val origin = URLDecoder.decode(matches.next().group(1), "utf-8")
+                if(matches.hasNext){
+                    val destination = URLDecoder.decode(matches.next().group(1), "utf-8")
+                    //print(origin + "-->" + destination)
```
Comment (dav009) [inline src/main/scala/org/idio/wikipedia/redirects/RedirectStore.scala]: comment

**Comment by dav009** [comment]
looks good :+1:  sorry for the late comment.
could you please squash some of the commits (for example: `local fixes` x 3)

**Comment by phdowling** [comment]
Ah, this is awful. I tried to squash and rebase, it seems to have just added more commits now

edit: looking better now. @dav009 feel free to review again and merge when ready

**Comment by dav009** [comment]
oops, `git rebase -i HEAD~20`  <- squashing.

**Comment by phdowling** [comment]
Is it looking okay now?

**Comment by phdowling** [comment]
@dav009 Just came across this again, any reason you don't want to merge? I think this improves the quality of the data a fair bit

**Comment by tgalery** [comment]
hi @phdowling @dav009 is in Japan atm, this repo is kind of not very well maintained as we might re-write it from scratch. That being said, I'll take a look at it when I have time, and probably merge.

**Pull request merged**